packages/fx-core/src/component/error/componentError.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  FxError,
+  SystemError,
+  SystemErrorOptions,
+  UserError,
+  UserErrorOptions,
+} from "@microsoft/teamsfx-api";
+import { getDefaultString, getLocalizedString } from "../../common/localizeUtils";
+import { DeployConstant } from "../constant/deployConstant";
+
+/**
+ * component error
+ * store more information in FxError's innerError property for telemetry, logging and debugging
+ *
+ * @property name name of error, for classification
+ * @property message always save error message in english
+ * @property errorType which type of error will throw to fx-core
+ * @property displayMessage the message show to user, always save as localized message
+ * @property helpLink link to help page, mostly no use
+ * @property detail detail error message, http response, console output, etc..
+ */
+export class BaseComponentInnerError extends Error {
+  errorType: "UserError" | "SystemError";
+  helpLink?: string;
+  displayMessage?: string;
+  suggestionKey?: string[];
+  detail?: string;
+
+  constructor(
+    errorType: "UserError" | "SystemError",
+    name: string,
+    messageKey?: string,
+    messageParams?: string[],
+    suggestionKey?: string[],
+    detail?: string,
+    helpLink?: string
+  ) {
+    super(
+      messageKey
+        ? messageParams
+          ? getDefaultString(messageKey, ...messageParams)
+          : getDefaultString(messageKey)
+        : undefined
+    );
+    this.errorType = errorType;
+    this.name = name;
+    this.helpLink = helpLink;
+    this.suggestionKey = suggestionKey;
+    this.displayMessage = messageKey
+      ? messageParams
+        ? getLocalizedString(messageKey, ...messageParams)
+        : getLocalizedString(messageKey)
+      : undefined;
+    this.detail = detail;
+  }
+
+  toFxError(): FxError {
+    if (this.errorType === "UserError") {
+      return new UserError({
+        source: "Deploy",
```

Comment:
It seems that L72 use constant, but here is not, can we keep them consistent to use constant?